docs/problem-matchers.md

```diff
@@ -0,0 +1,116 @@
+# Problem Matchers
+Problem Matchers are a way to scan the output of actions for a specified regex pattern and surface that information prominently in the UI. Both [GitHub Annotations](https://developer.github.com/v3/checks/runs/#annotations-object-1) and log file decorations are created when a match is detected.
+
+## Single Line Matchers
+
+Let's consider the ESLint compact output:
+```
+badFile.js: line 50, col 11, Error - 'myVar' is defined but never used. (no-unused-vars)
+```
+We can define a problem matcher in json that detects input in that format:
+```
+{
+    "problemMatcher": [
+        {
+            "owner": "eslint-compact",
+            "pattern": [
+                {
+                    // The pattern that provides groups to match against
+                    "regexp": "^(.+):\\sline\\s(\\d+),\\scol\\s(\\d+),\\s(Error|Warning|Info)\\s-\\s(.+)\\s\\((.+)\\)$",
+                    // The first group matches the file
+                    "file": 1,
+                    // The second group matches the line
+                    "line": 2,
+                    // The third group matches the column
+                    "column": 3,
+                    // The fourth group matches the severity
+                    "severity": 4,
+                    // The fifth group matches the message
+                    "message": 5,
+                    // The sixth group matches the code
+                    "code": 6
+                }
+            ]
+        }
+    ]
+}
+```
+
+The following fields are available for problem matchers:
+
+```
+{
+    owner: An id field that can be used to remove or replace the problem matcher. **required**
+    pattern: 
+    [
+        {
+            regexp: The regex containing the groups. **required**
+            file: a group number containing the file name
+            line: a group number containing the line number
+            column: a group number containing the column information
+            severity: a group number containing either 'warning' or 'error' case-insensitive. Defaults to `error`
+            code: a group number containing the error code
+            message: a group number containing the error message. **required** at least one pattern must set message
+            fromPath: a group number containing the base path of the file, otherwise the location of the git repository on disk is used
```

Comment:
\"base\" might be confusing (e.g. `basename /foo/bar.txt` outputs `bar.txt`)

Consider "root" or "root path" terminology. Note fromPath is expected to be a *file*. There is a bug in the runner i'm working on a fix now.